src/annotation/index.ts

```diff
@@ -751,6 +766,32 @@ function deserializeTwoFloatVectors(
   return offset;
 }
 
+function lineLength(
+  annotationLayerPositions: Float32Array<ArrayBufferLike>[],
+  scales: Float64Array,
+  units: readonly string[],
+) {
+  if (annotationLayerPositions.length !== 2) {
+    return;
+  }
+  const [pointA, pointB] = annotationLayerPositions;
+  const scalesRank = scales.length;
+  const lineRank = pointA.length;
+  if (scalesRank < lineRank) {
+    return;
+  }
+  let lengthSquared = 0;
+  for (let dim = 0; dim < lineRank; dim++) {
+    const unitInfo = ALLOWED_UNITS.find((x) => x.unit === units[dim]);
```

Comment:
`units` will always be normalized to "m" if it is some multiple of meters, so there is no need to check ALLOWED_UNITS.

Note that ALLOWED_UNITS is actually from some now-dead code that was originally used to show annotation properties just like you are adding with this PR.

However, you may as well support any unit for length --- just as long as all units for which there is a non-zero delta are the same.

Or perhaps it could be represented as  X meters + Y seconds, if for example there are both meters and seconds units.  You would compute the length separately for each unit.

The actual SI prefix can be selected using pickSiPrefix, as for the scale bar.